src/bind.py

```diff
@@ -114,3 +127,96 @@ def _install_snap_package(
             error_msg = f"An exception occurred when installing {snap_name}. Reason: {e}"
             logger.error(error_msg)
             raise InstallError(error_msg) from e
+
+    def _push(self, path: pathlib.Path, source: str) -> None:
+        """Pushes a file to the unit.
+
+        Args:
+            path: The path of the file
+            source: The contents of the file to be pushed
+        """
+        with open(path, "w", encoding="utf-8") as write_file:
+            write_file.write(source)
+            logger.info("Pushed file %s", path)
+
+    def _to_bind_zones(self, rrd: DNSRecordRequirerData) -> typing.Dict[str, str]:
+        """Convert DNSRecordRequirerData to zone files.
+
+        Args:
+            rrd: The input DNSRecordRequirerData
+
+        Returns:
+            A dict of zones names as keys with the zones contents as values
+        """
+        zones_entries: typing.Dict[str, typing.List[RequirerEntry]] = {}
+        for entry in rrd.dns_entries:
+            if entry.domain not in zones_entries:
+                zones_entries[entry.domain] = []
+            zones_entries[entry.domain].append(entry)
+
+        zones_content: typing.Dict[str, str] = {}
+        for zone, entries in zones_entries.items():
+            content: str = (
+                f"$ORIGIN {zone}.\n"
```

Comment:
Perhaps this template should be a constant?

Reply:
I can't store an f-string for later use as it has to be evaluated at creation. If our needs grow, we could use jinja.

Reply:
You can define the string as a constant and then evaluate it later

Reply:
Understood. I used `str.format()` as a lightweight templating system